### PR TITLE
Store channel keys

### LINKED ISSUE #412: Join keys are not stored
https://tools.ietf.org/html/rfc2812#section-3.2.1

We need to store them in `Chan` object and need to write them to user config to correctly rejoin the channels. To fix this in the user config, I think we need to deprecate the `join` string, and store the `channels` array properly (only name and join keys).

### PR DESCRIPTION
**Pull request by xPaw**
Fixes #412 and supercedes #715. This PR gets channel key from channel mode, instead of hooking join command.

cc @spookhurb, @DanielOaks

### CODE REVIEW

**Comment by astorije** [comment]
Just like the parser PR, I think the commit author should be @spookhurb :)

**Comment by xPaw** [comment]
I didn't re-use his code (besides test), and it uses a different approach.

File: src/plugins/irc-events/mode.js [hunk 1]
```diff
@@ -6,6 +6,29 @@ var Msg = require("../../models/msg");
...
+			return;
+		}
+
+		const targetChan = network.getChannel(data.channel);
+		if (typeof targetChan === "undefined") {
+			return;
+		}
+
+		for (var i = 0; i < data.modes.length; i++) {
+			var mode = data.modes[i];
```
Comment (astorije) [inline src/plugins/irc-events/mode.js]: This is a minor nitpick, but any reason why not to use `data.modes.forEach(mode => { ... });` instead?
Also, `const` when possible?

Comment (xPaw) [inline src/plugins/irc-events/mode.js]: Done.

**Pull request merged**